charts-external/spark/templates/spark-configmap.yaml

```diff
@@ -59,4 +59,7 @@ data:
   {{ if .Values.volunteersBaseUrl }}
   VOLUNTEERS_BASE_URL: {{ .Values.volunteersBaseUrl | quote }}
   {{ end }}
+
+  # SUPPLIER CONTRACT FILES
+  SPARK_SUPPLIER_CONTRACTS_BUCKET: {{ .Values.suppliers.bucket | quote }}
```

Comment:
each chart has access only to values under it's name (and the `global` property)

so, in your case it should be under
{{ .Values.suppliersBucket | quote }}

also, you should check if value exists before adding it - see the example of volunteersBaseUrl above your code